src/shared/proto/robot_component_status.proto

```diff
@@ -0,0 +1,132 @@
+syntax = "proto2";
+
+/* Data about the status of the break beam */
+message BreakBeamStatus
+{
+    // Whether or not the ball is interrupting the robot’s laser beam.
+    required bool ball_in_beam = 1;
+
+    // The reading of the robot’s laser sensor.
+    required float break_beam_reading = 2;
+}
+
+/* Data about the status of the chipper and kicker */
+message ChipperKickerStatus
+{
+    /* Frankie Robot Fields */
+    // Indicates the time elapsed since chipper was last fired in ms
+    // UINT64_MAX indicates chipper was never fired
+    optional uint64 time_since_chipper_fired = 2;
+
+    // Indicates the time elapsed since kicker was last fired in ms
+    // UINT64_MAX indicates kicker was never fired
+    optional uint64 time_since_kicker_fired = 3;
+
+    /* 2014 Robot Fields */
+    // Indicates when autokick has been fired.
+    optional bool autokick_fired = 8;
+}
+
+/* Data about the status of the dribbler */
+message DribblerStatus
+{
+    optional int32 dribbler_rpm = 1;
+}
+
+/* Error codes are to be displayed in the visualizer */
+enum ErrorCode
+{
+    /** 2014 ROBOT Messages **/
+    // Electrical messages
```

Comment:
nit: I don't think we need the `Electrical Messages` heading since a lot more of these errors are electrical. I think splitting by level-triggered and edge-triggered is more important.

nit 2: Most of these error codes apply to "all" robots, so I also don't think we need the `2014 Robot messages` header

Reply:
done